src/app/page.tsx

```diff
@@ -1,12 +1,15 @@
 import Wrapper from '@components/map/wrapper';
 import { AppContextProvider } from '@contexts/app-context';
 import Drawer from '@components/menu/drawer';
+import ThemeContextProvider from '@contexts/theme-context';
 
 export default function Home() {
   return (
-    <AppContextProvider>
-      <Wrapper />
-      <Drawer />
-    </AppContextProvider>
+    <ThemeContextProvider>
+      <AppContextProvider>
+        <Wrapper />
+        <Drawer />
+      </AppContextProvider>
+    </ThemeContextProvider>
```

Comment:
Please place the logic for switching themes in `AppContextProvider` instead of creating a new context provider `ThemeContextProvider` 

```suggestion
    <AppContextProvider>
      <ThemeProvider theme={/* insert theme value from app context here */} >
        <Wrapper />
        <Drawer />
      </ThemeProvider>
    </AppContextProvider>
```